test/unit/event/send-acknowledgement-event.test.js

```diff
@@ -0,0 +1,56 @@
+
+jest.mock('../../../app/event/send-processing-ack-error-event')
+const sendProcessingAckErrorEvent = require('../../../app/event/send-processing-ack-error-event')
+
+jest.mock('../../../app/event/send-invalid-bank-details-event')
+const sendInvalidBankDetailsEvent = require('../../../app/event/send-invalid-bank-details-event')
+
+let mockAcknowledgementError
+let mockHoldCategoryName
+
+describe('send acknowledgement event', () => {
+  beforeEach(() => {
+    mockAcknowledgementError = require('../../mock-acknowledgement-error')
+    mockHoldCategoryName = 'DAX rejection'
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+    jest.resetModules()
+  })
+
+  test('should call sendProcessingAckErrorEvent when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalled()
+  })
+
+  test('should call sendProcessingAckErrorEvent once when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call sendProcessingAckErrorEvent with the unsuccessful ack object when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "DAX rejection"', async () => {
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendProcessingAckErrorEvent).toHaveBeenCalledWith(mockAcknowledgementError)
+  })
+
+  test('should call sendInvalidBankDetailsEvent when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "Bank account anomaly"', async () => {
+    mockHoldCategoryName = 'Bank account anomaly'
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendInvalidBankDetailsEvent).toHaveBeenCalled()
+  })
+
+  test('should call sendInvalidBankDetailsEvent once when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "Bank account anomaly"', async () => {
+    mockHoldCategoryName = 'Bank account anomaly'
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendInvalidBankDetailsEvent).toHaveBeenCalledTimes(1)
+  })
+
+  test('should call sendInvalidBankDetailsEvent with the unsuccessful ack object when a holdCategoryName, an unsuccessful ack object and frn is given and holdCategoryName is "Bank account anomaly"', async () => {
+    mockHoldCategoryName = 'Bank account anomaly'
+    await sendAcknowledgementEvent(mockHoldCategoryName, mockAcknowledgementError, mockFRN)
+    expect(sendInvalidBankDetailsEvent).toHaveBeenCalledWith(mockFRN)
+  })
+})
```

Comment:
Could you add tests for:
- `sendInvalidBankDetailsEvent` throws
- `sendProcessingAckErrorEvent` throws

Reply:
What do you mean by throws here?

Reply:
Throws an exception, i.e. breaks.

We can use the `jest` function mockRejectedValue(new Error('Suitable error name/message'))

Means we have to wrap our function call up and check the wrapped call does or does not throw itself, i.e. it won't throw if a `try-catch` is used and in our `catch` we don't raise an exception, e.g. https://github.com/DEFRA/ffc-pay-statement-constructor/blob/main/test/unit/inbound/process-organisation.test.js#L76